src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,526 @@
+            <HttpConnectionResponseInterceptor>();
+    private String password;
+    private String username;
+    private URL url;
+    private GsonBuilder gsonBuilder;
+    /**
+     * Defaults to {@link #DEFAULT_MAX_CONNECTIONS}
+     **/
+    private int maxConnections = DEFAULT_MAX_CONNECTIONS;
+    private URL proxyURL;
+    private String proxyUser;
+    private String proxyPassword;
+    private boolean isSSLAuthenticationDisabled;
+    private SSLSocketFactory authenticatedModeSSLSocketFactory;
+    private TimeoutOption connectionTimeout = DEFAULT_CONNECTION_TIMEOUT;
+    private TimeoutOption readTimeout = DEFAULT_READ_TIMEOUT;
+
+    /**
+     * Constructs a new ClientBuilder for building a CloudantClient instance to connect to the
+     * Cloudant server with the specified account.
+     *
+     * @param account the Cloudant account name to connect to e.g. "example" is the account name
+     *                for the "example.cloudant.com" endpoint
+     * @return a new ClientBuilder for the account
+     * @throws MalformedURLException if the specified account name forms an invalid endpoint URL
+     */
+    public static ClientBuilder account(String account) throws MalformedURLException {
+        return new ClientBuilder(account);
+    }
+
+    /**
+     * Constructs a new ClientBuilder for building a CloudantClient instance to connect to the
+     * Cloudant server with the specified URL.
+     *
+     * @param url server URL e.g. "https://yourCloudantLocalAddress.example"
+     * @return a new ClientBuilder for the account
+     */
+    public static ClientBuilder url(URL url) {
+        return new ClientBuilder(url);
+    }
+
+    private ClientBuilder(String account) throws MalformedURLException {
+        //parse account into a url
+        assertNotEmpty(account, "accountName");
+        url(URIBuilder.buildUri()
+                .scheme("https")
+                .host(account + ".cloudant.com")
+                .build().toURL());
+    }
+
+
+    private ClientBuilder(URL url) {
+        //Check if port exists
+        int port = url.getPort();
+        if (port < 0) {
+            port = url.getDefaultPort();
+        }
+        if (url.getUserInfo() != null) {
+            //Get username and password and replace credential variables
+            this.username = url.getUserInfo().substring(0, url
+                    .getUserInfo()
+                    .indexOf(":"));
+            this.password = url.getUserInfo().substring(url
+                    .getUserInfo()
+                    .indexOf(":"));
+        }
+
+        try {
+            //Remove user credentials from url
+            this.url = URIBuilder.buildUri()
+                    .scheme(url.getProtocol())
+                    .host(url.getHost())
+                    .port(port)
+                    .build().toURL();
+        } catch (MalformedURLException e) {
+            throw new CouchDbException(e);
```

Comment:
As part of the bigger picture to eventually have checked exceptions, I think this should be specifically a runtime exception. We shouldn't hit the MalformedURLException, since the url is the same that was provided to us, minus the user info. So I think we should just throw a RuntimeException directly.